pkg/database/resync.go

```diff
@@ -4,45 +4,275 @@ package database
 
 import (
 	"context"
+	"encoding/json"
 	"fmt"
+	"reflect"
+	"time"
 
 	"github.com/stolostron/search-indexer/pkg/metrics"
 	"github.com/stolostron/search-indexer/pkg/model"
 	"k8s.io/klog/v2"
 )
 
-// Overrides the existing state of a cluster with the new data.
-// NOTE: This logic is not optimized. We use the simplest approach because this is a failsafe to
-//       recover from rare sync problems. At the moment this is good enough without adding complexity.
+// Reset data for the cluster to the incoming state.
 func (dao *DAO) ResyncData(ctx context.Context, event model.SyncEvent,
 	clusterName string, syncResponse *model.SyncResponse) error {
 
-	defer metrics.SlowLog(fmt.Sprintf("Slow Resync from cluster %s", clusterName), 0)()
+	defer metrics.SlowLog(fmt.Sprintf("Slow resync from cluster %s", clusterName), 0)()
```

Comment:
Should we print the request id in the error messages to track the size of the request with collector?

Reply:
updated